service/src/main/java/org/ehrbase/service/KnowledgeCacheService.java

```diff
@@ -159,19 +160,31 @@ public Set<String> getAllTemplateIds() {
   }
 
   @Override
-  public String addOperationalTemplate(byte[] content) {
-    return addOperationalTemplateIntern(content, false);
+  public String addOperationalTemplate(InputStream inputStream) {
+
+    OPERATIONALTEMPLATE template = buildOperationalTemplate(inputStream);
+
+    return addOperationalTemplateIntern(template, false);
   }
 
-  public String addOperationalTemplateIntern(byte[] content, boolean overwrite) {
+  private OPERATIONALTEMPLATE buildOperationalTemplate(InputStream content) {
```

Comment:
It seems most of the callers fail to close the stream (try-with resources).
Alternatively the method could accept a Consumer<InputStream> and take care of closing it.

Reply:
since all callers are unit tests i would just leave this as is, with the hope that if it is used in new code the input stream is properly handled

Reply:
Reluctantly agreeing :)